components/admin_console/system_users/list/system_users_list.jsx

```diff
@@ -68,10 +70,13 @@ export default class SystemUsersList extends React.Component {
         };
     }
 
-    componentDidUpdate(prevProps) { // eslint-disable-line camelcase
-        if (prevProps.teamId !== this.props.teamId || prevProps.filter !== this.props.filter) {
-            this.setState({page: 0});
+    static getDerivedStateFromProps(nextProps, prevState) {
+        if (prevState.teamId !== nextProps.teamId || prevState.filter !== nextProps.filter) {
+            return {
+                page: 0,
```

Comment:
You also need to update the `teamId` and `filter` here. Currently, you have them set to their initial values based on the state, but they're never updated after that. If you change this to
```
return {
    page: 0,
    teamId: nextProps.teamId,
    filter: nextProps.filter,
};
```
this should be good to go